### PR TITLE
issue/1

### LINKED ISSUE #1: Support Safari's fullscreen API
Really thought they would have caught up on fullscreen API naming by now...

`el.requestFullscreen` does not work. Safari has `el.webkitRequestFullScreen`.

Which also means the other fullscreen related APIs are going to be different still.

### PR DESCRIPTION
**Pull request by littlespex**
Added support for legacy fullscreen APIs

Resolves #1

### CODE REVIEW

File: src/js/media-fullscreen-button.js [hunk 1]
```diff
@@ -20,30 +20,69 @@ class MediaFullscreenButton extends MediaChromeButton {
...
-      const media = this.media;
-
-      if (this.mediaChrome == document.fullscreenElement) {
-        document.exitFullscreen();
-      } else {
-        if (document.pictureInPictureElement) {
-          // Should be async
-          document.exitPictureInPicture();
-        }
+  onClick() {
```
Comment (littlespex) [inline src/js/media-fullscreen-button.js:25]: Moved the button click functionality to the `onClick` method provided by the base class.

File: src/js/media-fullscreen-button.js [hunk 2]
```diff
@@ -20,30 +20,69 @@ class MediaFullscreenButton extends MediaChromeButton {
...
-      if (this.mediaChrome == document.fullscreenElement) {
+      api.enter();
+    }
+  }
+
+  mediaSetCallback() {
+    const api = this.getFullscreenApi(this.mediaChrome);
+    
+    document.addEventListener(api.event, () => {
+      if (this.mediaChrome == document[api.element]) {
```
Comment (littlespex) [inline src/js/media-fullscreen-button.js:44]: The `getFullscreenApi` method needs a reference to the `media-chrome` element, so the fullscreen event listener has been moved to the `mediaSetCallback`

File: src/js/media-fullscreen-button.js [hunk 3]
```diff
@@ -20,30 +20,69 @@ class MediaFullscreenButton extends MediaChromeButton {
...
+    else if (container.msRequestFullscreen != null) {
+      api.enter = container.msRequestFullscreen.bind(container);
+      api.exit = document.msExitFullscreen.bind(document);
+      api.event = "msfullscreenchange";
+      api.element = "msFullscreenElement";
+      api.error = "MSFullscreenError";
+    }
+
+    return api;
+  }
```
Comment (littlespex) [inline src/js/media-fullscreen-button.js:85]: Hooray for standards...

**Review by heff — APPROVED** [review]
Added a fix for this issue

image

File: src/js/media-chrome.js [hunk 4]
```diff
@@ -18,6 +17,12 @@ template.innerHTML = `
       flex-direction: column-reverse;
     }
 
+    :host(:-webkit-full-screen) {
```
Comment (heff) [inline src/js/media-chrome.js:20]: This was also needed for Safari...

File: src/js/media-fullscreen-button.js (hunk 3 above)
Comment (littlespex) [reply src/js/media-fullscreen-button.js:85]: If the baseline is "browsers that support custom elements" I think we could remove the MS section as well.

**Pull request merged**